crypto/stateproof/builder.go

```diff
@@ -258,3 +262,8 @@ func (b *Builder) Build() (*StateProof, error) {
 	b.cachedProof = s
 	return s, nil
 }
+
+// AllocSigs should only be used after decoding msgpacked Builder, as the sigs field is not exported and encoded
+func (b *Builder) AllocSigs() {
```

Comment:
Why not insert it into the msg-pack code? You can use these kinds of directives: 
You might be able to do it in the msgpack code by using post-unmarshal callbacks.

//msgp:postunmarshalcheck {Type} {funcName} {funcName} 
added here: https://github.com/algonathan/msgp/commit/9359fd17b7acb7fa5f9a8d1d78336668b9f9d129